### DISCUSSION TITLE
How to plot the Young's modulus of aluminum alloy

### DISCUSSION THREAD

**Discussion by xiu-shen** [mtex-toolbox/mtex#2241] [Ask Anything]
```matlab
fname = fullfile(mtexDataPath,'tensor','Olivine1997PC.GPa');

C = stiffnessTensor.load(fname,ebsd('Aluminium').CS)
```

%%
警告: Aluminium Tensor does not pose the right symmetry 
> 位置：tensor/symmetrise (第 87 行)
位置: tensor (第 139 行)
位置: loadTensor_generic (第 35 行)
位置: check_interfaces (第 47 行)
位置: tensor.load (第 24 行)
位置: stiffnessTensor.load (第 23 行) 
Deviation:70
警告: Aluminium Tensor does not pose the right symmetry 
> 位置：tensor/symmetrise (第 87 行)
位置: tensor (第 139 行)
位置: loadTensor_generic (第 35 行)
位置: tensor.load (第 32 行)
位置: stiffnessTensor.load (第 23 行)

**Comment by lfgmorales** [comment]
Hi

You need the elastic constant for aluminum plus the crystal symmetry of that material for your command to work

Here is how the elastic constant should look like:

<https://mtex-toolbox.github.io/CPOSeismicProperties.html>
Plot seismic wave velocities and polarization directions for aggregates | MTEX<https://mtex-toolbox.github.io/CPOSeismicProperties.html>
mtex-toolbox.github.io<https://mtex-toolbox.github.io/CPOSeismicProperties.html>
        [favicon.png] <https://mtex-toolbox.github.io/CPOSeismicProperties.html>



The elastic constant for your material you can find published

Cheers

Luiz


Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 8 Oct 2024, at 05:22, xiu-shen ***@***.***> wrote:

﻿

fname = fullfile(mtexDataPath,'tensor','Olivine1997PC.GPa');

C = stiffnessTensor.load(fname,ebsd('Aluminium').CS)

%%
警告: Aluminium Tensor does not pose the right symmetry

位置：tensor/symmetrise (第 87 行)
位置: tensor (第 139 行)
位置: loadTensor_generic (第 35 行)
位置: check_interfaces (第 47 行)
位置: tensor.load (第 24 行)
位置: stiffnessTensor.load (第 23 行)
Deviation:70
警告: Aluminium Tensor does not pose the right symmetry
位置：tensor/symmetrise (第 87 行)
位置: tensor (第 139 行)
位置: loadTensor_generic (第 35 行)
位置: tensor.load (第 32 行)
位置: stiffnessTensor.load (第 23 行)

**Reply by xiu-shen** [reply]
I looked it up on the Internet

The adiabatic elastic constants of aluminum have been measured with an ultrasonic pulse technique at liquid‐heluim, liquid‐nitrogen, and room temperatures. The values of the elastic constants at 4°K are c11=11.63, c12=6.48, and c44=3.09 in units of 1010 N/m2. The Debye characteristic temperature calculated from these figures is θ=425°K in good agreement with the value obtained from speciffic heat data.The shear elastic constants ½(c11−c12) and c44 are discussed with reference to the

After conversion does not get the following formula
Cij = [[11.63  6.48  0    0     0     0];...
          [ 6.48   0     0     0     0     0];...
          [  0      0     0      0     0     0];...
          [   0     0    0   3.09   0   0];...
          [   0     0    0      0   0    0];...
          [   0     0    0      0   0   0]];

**Comment by lfgmorales** [comment]
Maybe you need to look to some basic concepts symmetry in elastic constants, symmetry related to stress and strain, crystal symmetry etc, your matrix is incomplete and wrong

Something like

<https://commons.m.wikimedia.org/wiki/File:Cubic_elastic_constants.JPG>[Cubic_elastic_constants.JPG]
File:Cubic elastic constants..JPG - Wikimedia Commons<https://upload.wikimedia.org/wikipedia/commons/4/40/Cubic_Elastic_Constants.JPG>
wikimedia.org<https://commons.m.wikimedia.org/wiki/File:Cubic_elastic_constants..JPG>

Cheers

Luiz


Luiz F. G. Morales
ScopeM - ETH Zürich

**Reply by xiu-shen** [reply]
I rewrote this matrix. Do you see if this matrix is right? Thank you very much for your help
Cij = [[11.63 6.48   6.48     0      0       0];...
         [ 6.48  11.63   6.48    0      0      0];...
         [ 6.48  6.48   11.63    0      0      0];...
         [   0       0        0     3.09    0      0];...
         [   0       0        0        0    3.09   0];...
         [   0       0        0        0     0   3.09];...

**Comment by lfgmorales** [comment]
Hi

Now it looks fine

Cheers

Luiz
